crates/nargo_cli/src/manifest.rs

```diff
@@ -1,13 +1,234 @@
+    // For now, we state that all packages must be compiled under the same
+    // compiler version.
+    // We also state that ACIR and the compiler will upgrade in lockstep.
+    // so you will not need to supply an ACIR and compiler version
+    compiler_version: Option<String>,
+    backend: Option<String>,
+    license: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[serde(untagged)]
+/// Enum representing the different types of ways to
+/// supply a source for the dependency
+pub(super) enum Dependency {
+    Github { git: String, tag: String },
+    Path { path: String },
+}
+
+impl PackageConfig {
+    fn to_default_package(&self, root_path: &Path) -> Result<Package, ManifestError> {
+        let name = match self.package.name.as_ref() {
+            Some(name) => {
+                Some(CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?)
+            }
+            None => None,
+        };
+
+        let mut dependencies: BTreeMap<CrateName, package::Dependency> = BTreeMap::new();
+        for (name, dep) in self.dependencies.iter() {
+            let name = CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?;
+            let resolved_dep = resolve_dependency(root_path, dep)?;
+
+            dependencies.insert(name, resolved_dep);
+        }
+
+        Ok(Package { root_path: root_path.to_path_buf(), is_default: true, name, dependencies })
+    }
+    fn to_package(&self, root_path: &Path) -> Result<Package, ManifestError> {
+        let name = match self.package.name.as_ref() {
+            Some(name) => {
+                Some(CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?)
+            }
+            None => None,
+        };
+
+        let mut dependencies: BTreeMap<CrateName, package::Dependency> = BTreeMap::new();
+        for (name, dep) in self.dependencies.iter() {
+            let name = CrateName::new(name).map_err(|_| ManifestError::InvalidPackageName)?;
+            let resolved_dep = resolve_dependency(root_path, dep)?;
+
+            dependencies.insert(name, resolved_dep);
+        }
+
+        Ok(Package { root_path: root_path.to_path_buf(), is_default: false, name, dependencies })
+    }
+}
+
+fn from_file(root_path: &Path) -> Result<NargoToml, ManifestError> {
     let toml_as_string =
-        std::fs::read_to_string(&path_to_toml).expect("ice: path given for toml file is invalid");
+        std::fs::read_to_string(root_path).expect("ice: path given for toml file is invalid");
+    let manifest = toml_as_string.try_into()?;
+    Ok(manifest)
+}
+
+pub(super) fn resolve_workspace(toml_path: &Path) -> Result<Workspace, ManifestError> {
+    let root_path = toml_path.parent().expect("TODO");
+    let manifest = from_file(toml_path)?;
+
+    let workspace = match manifest {
+        NargoToml::Package(pkg_config) => {
+            let member = pkg_config.to_default_package(root_path)?;
+            Workspace { members: vec![member] }
+        }
```

Comment:
We'll want to check for a workspace in the parent directories even if we're already in the package so we can resolve workspace dependencies, etc. in future.

Reply:
I'd like to defer this one. I think this needs further design consideration; for example, we wouldn't need to search for & parse a workspace manifest if a Nargo.toml didn't include any dependencies that were `.workspace = true`

Reply:
Yeah, I'm good with deferring this as it opens up new error types which we'd need to handle, e.g. package which sits inside another package rather than a workspace, etc.